ast/src/parsed/display.rs

```diff
@@ -549,7 +595,13 @@ impl<Ref: Display> Display for Expression<Ref> {
             Expression::Tuple(items) => write!(f, "({})", format_list(items)),
             Expression::LambdaExpression(lambda) => write!(f, "{}", lambda),
             Expression::ArrayLiteral(array) => write!(f, "{array}"),
-            Expression::BinaryOperation(left, op, right) => write!(f, "({left} {op} {right})"),
+            Expression::BinaryOperation(left, op, right) => write!(
```

Comment:
Before this change, this always had parentheses. Now the outer structure never has parentheses. Shouldn't we rather always add parentheses unless we know the precedence of the outer structure?